app/assets/javascripts/galleries/add_new.js

```diff
@@ -1,44 +1,38 @@
 /* global deleteUnusedIcons */
+/* exported addUploadedIcon, addCallback, failCallback */
+
 var done = 0;
 var total = 0;
 var failed = 0;
 
 $(document).ready(function() {
   fixButtons();
   $(".icon-row td:has(input)").each(function() {
-    $(this).keydown(processDirectionalKey);
+    $(this).keydown(function(event) {
+      if (event.which < 37 || event.which > 40) { return; } // skip if not a directional key
```

Comment:
(Forced braces are weird and these are magic numbers – can we use the .indexOf thing like before?)

Reply:
yeah I pulled these to here because of the complexity complaint and didn't want to redeclare the variables but I could make a [37, 38, 39, 40] list probably?

Or, yes, we could ignore it, but it wasn't *wrong* that that function was hideously complicated.

Reply:
I'd move the 37, 38, 39, 40 magic numbers to a global constant and refer to them from both functions